src/stream/src/executor/mview/materialize.rs

```diff
@@ -65,13 +65,44 @@ pub struct ArrangeExecutorBuilder;
 
 impl ExecutorBuilder for ArrangeExecutorBuilder {
     fn new_boxed_executor(
-        _params: ExecutorParams,
+        mut params: ExecutorParams,
         node: &stream_plan::StreamNode,
-        _store: impl StateStore,
+        store: impl StateStore,
         _stream: &mut LocalStreamManagerCore,
     ) -> Result<Box<dyn Executor>> {
-        let _node = try_match_expand!(node.get_node().unwrap(), Node::ArrangeNode)?;
+        let arrange_node = try_match_expand!(node.get_node().unwrap(), Node::ArrangeNode)?;
```

Comment:
Why introduce `ArrangeNode` besides `MaterializeNode`?

Reply:
Arrange node has simpler information than a typical materialize node. We only need to provide "arrange keys". On the compute node side, it will automatically generate a sequence of column ids and column descs for MaterializeExecutor to use.

Reply:
Also this is an example that materialize node is using different distribution from the materialize keys. For MaterializeExecutor constructed from MaterializeNode, its distribution key is always the same as pks. For MaterializeExecutor constructed from ArrangeNode, the distribution key is join key, but and materialize key is join key + pk.

Reply:
Indeed. cc. @xx01cyx

Reply:
Got it.